weavermetadata/metadata.go

```diff
@@ -0,0 +1,67 @@
+// Copyright 2024 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package weavermetadata define the structure of the metadata supported by weaver.
+//
+// Note that we allow metadata information to propagate as a map, where the keys
+// and the values are strings. It the user wants to propagate values that are
+// more complicated types, it's their responsibility to encode/decode these types
+// to/from string values.
+//
+// Note that all keys are automatically lowercase. This ensures the user avoids
```

Comment:
We should drop this lower-casing I think.

Reply:
Done